### PR TITLE
[expo-go] Add method to request local network access

### PR DESCRIPTION
**Pull request by alanjhughes**
# Why
We need a way to trigger the local network permission dialog on iOS before a user attempts to load an app. It is requested when we attempt to connect to the dev server socket but this is too late as RN does not wait for the result and will hard crash when it can't connect. 

# How
I found a reliable method [here](https://gist.github.com/mac-cain13/fa684f54a7ae1bba8669e78d28611784) that will also give us the result of the request. We need to update the plist to add bonjour but I don't believe this requires any explanation to apple. We can make further enhancements where we do this before attempting to load any app and can provide appropriate messaging if the permission is denied. For now, we trigger the dialog once the app becomes active. 

# Test Plan
Expo go. A fresh install will request the permission on launch. Subsequent launches do nothing.

### CODE REVIEW

**Review by lukmccall — APPROVED** [review]
If that's a reliable way of requesting the local network access, we probably should add it to the dev-launcher too.

File: apps/expo-go/ios/Exponent.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1280,9 +1283,7 @@
 				TargetAttributes = {
 					78CEE2BF1ACD07D70095B124 = {
 						CreatedOnToolsVersion = 6.2;
-						DevelopmentTeam = C8D8QTF339;
 						LastSwiftMigration = 1150;
-						ProvisioningStyle = Automatic;
```
Comment (Kudo) [inline apps/expo-go/ios/Exponent.xcodeproj/project.pbxproj:1285]: maybe revert these changes

File: apps/expo-go/ios/Exponent.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -2158,8 +2154,8 @@
 				CLANG_ALLOW_NON_MODULAR_INCLUDES_IN_FRAMEWORK_MODULES = YES;
 				CLANG_ENABLE_MODULES = YES;
 				CODE_SIGN_ENTITLEMENTS = Exponent/Supporting/Exponent.entitlements;
-				CODE_SIGN_IDENTITY = "Apple Distribution";
-				CODE_SIGN_STYLE = Manual;
```
Comment (Kudo) [inline apps/expo-go/ios/Exponent.xcodeproj/project.pbxproj:2162]: maybe revert these changes

File: apps/expo-go/ios/Exponent.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -2203,7 +2199,7 @@
 				OTHER_SWIFT_FLAGS = "$(inherited) -D COCOAPODS -Xcc -fmodule-map-file=\"${PODS_ROOT}/Headers/Public/yoga/Yoga.modulemap\" -D RELEASE -D EXPO_CONFIGURATION_RELEASE";
 				PRODUCT_BUNDLE_IDENTIFIER = host.exp.Exponent;
 				PRODUCT_NAME = "Expo Go";
-				PROVISIONING_PROFILE_SPECIFIER = "match AppStore host.exp.Exponent";
```
Comment (Kudo) [inline apps/expo-go/ios/Exponent.xcodeproj/project.pbxproj:2206]: maybe revert these changes
